Junimatic/IndoorPotMachine.cs

```diff
@@ -0,0 +1,226 @@
+            {
+                if (JsonConvert.DeserializeObject<List<OutputObjectMetadata>>(HarvestObjectsString) is List<OutputObjectMetadata> HarvestObjectsMetadata)
+                    this.HarvestObjects = HarvestObjectsMetadata;
+            }
+        }
+
+        public new IndoorPot Machine => (IndoorPot)base.Machine;
+
+        public override bool IsIdle => false; // Only supporting harvesting
+
+        public override SObject? HeldObject => this.GetHeldObject();
+
+        private readonly List<OutputObjectMetadata> HarvestObjects = [];
+        private readonly Farmer FakeFarmer;
+        private readonly SObject DummyObject;
+
+        public override bool IsCompatibleWithJunimo(JunimoType projectType)
+        {
+            return projectType == JunimoType.Crops;
+        }
+
+        /// <summary>
+        /// Pull the next object from HarvestObjects or harvest the pot if HarvestObjects is not populated
+        /// </summary>
+        /// <returns>StardewValley.Object harvested from the pot</returns>
+        protected override SObject TakeItemFromMachine()
+        {
+            SObject HarvestObject = this.DummyObject; // Set to dummy object to prevent the game from crashing if harvest fails
+            // Harvest if HarvestObjects is not populated and pot is still ready to harvest
+            if (!this.HarvestObjects.Any() && this.IsHarvestable())
+            {
+                this.Harvest();
+                
+                // Log an error if harvest fails. This won't stop Junimatic from repeatedly trying to harvest it. (TODO)
+                if (!this.HarvestObjects.Any()) ModEntry.Instance.LogErrorOnce($"Machine failed to harvest at {this.Machine.TileLocation}");
+            }
+            // Return and remove the first object in HarvestObjects, if populated
+            if (this.HarvestObjects.Any())
+            {
+                HarvestObject = this.HarvestObjects[0].ConvertToSObject();
+                this.HarvestObjects.RemoveAt(0);
+
+                // Serialize the updated list to modData
+                // This fixes the issue where not all objects in HarvestObjects are retrieved as this class does not persist after this method is called.
+                // I.E. This class is recreated for the machine each time the junimo is looking for work
+                this.Machine.modData[IF($"{ModEntry.Instance.ModManifest.UniqueID}/HarvestObjects")] = JsonConvert.SerializeObject(this.HarvestObjects);
+            }
+            return HarvestObject;
+            
+        }
+
+        /// <summary>
+        /// Return the DummyObject to flag the pot as ready to empty if pot is ready to harvest or if there are still objects to be collected from HarvestObjects
+        /// </summary>
+        /// <returns>The DummyObject. Default: null</returns>
+        private SObject? GetHeldObject()
+        {
+            if (this.IsHarvestable() || this.HarvestObjects.Any()) return this.DummyObject;
```

Comment:
Your assumption that HeldObject is only used to test if there's an output isn't correct - see GameMachine.TryPutHeldObjectInStorage.  I think perhaps you tested this in "animated" mode, where the Junimos turn up and do their thing.  There's a whole parallel execution path for the case where the player is not present.